lesson-11/src/controllers/tasks.ts

```diff
@@ -0,0 +1,116 @@
+import { Request, Response } from 'express';
+import { z } from 'zod';
+import * as taskService from '../services/tasks';
+import { TaskFilters } from '../types/tasks';
+
+const createTaskSchema = z.object({
+  title: z.string().min(1),
+  description: z.string().optional(),
+  status: z.enum(['todo', 'in_progress', 'done']).optional(),
+  priority: z.enum(['low', 'medium', 'high']).optional()
+});
+
+const updateTaskSchema = createTaskSchema.partial();
+
+const querySchema = z.object({
+  createdAt: z.string().optional(),
+  status: z.string().optional(), // CSV or single
+  priority: z.string().optional() // CSV or single
+});
+
+function parseCsv(value?: string): string[] | undefined {
+  if (!value) return undefined;
+  return value.split(',').map(s => s.trim()).filter(Boolean);
+}
+
+export const getAllTasks = (req: Request, res: Response) => {
+  try {
+    const parseResult = querySchema.safeParse(req.query);
+    if (!parseResult.success) {
+      return res.status(400).json({ message: 'Invalid query params', errors: parseResult.error.issues });
+    }
+
+    const { createdAt, status, priority } = parseResult.data;
+    const filters: TaskFilters = {};
+    if (createdAt) filters.createdAt = createdAt;
+    if (status) filters.status = parseCsv(status) as any;
+    if (priority) filters.priority = parseCsv(priority) as any;
+
+    const result = taskService.getAllTasks(filters);
+    res.json(result);
+  } catch (err) {
+    console.error(err);
+    res.status(500).json({ message: 'Internal Server Error' });
+  }
+};
+
+export const getTask = (req: Request, res: Response) => {
+  try {
+    const id = req.params.id as string;
+    const task = taskService.getTaskById(id);
+    if (!task) return res.status(404).json({ message: 'Task not found' });
```

Comment:
замість того щоб повертати помилку у кожному методі, ти можеш передавати її у global error handler, який в тебе вже реалізований
```suggestion
    if (!task) throw new ApiError('Task not found', 404);
```